src/info/mod.rs

```diff
@@ -304,27 +304,58 @@ impl Info {
     }
 
     fn get_language_field(&self, title: &str) -> String {
-        let mut language_field = String::from("");
-
+        let language_bar_length = 26;
         let pad = title.len() + 2;
+        let color_palette = vec![
+            Color::Red,
+            Color::Green,
+            Color::Yellow,
+            Color::Blue,
+            Color::Magenta,
+            Color::Cyan,
+        ];
 
-        let languages: Vec<(String, f64)> = {
-            let mut iter = self.languages.iter().map(|x| (format!("{}", x.0), x.1));
+        let mut language_field = String::from("");
+        let mut i = 0;
+        let languages: Vec<(String, f64, Color)> = {
+            let mut iter = self.languages.iter().map(|x| {
+                i += 1;
+                let color = if self.config.true_color {
+                    x.0.get_colors(true)[0]
+                } else {
+                    color_palette[i % color_palette.len()]
+                };
+                (format!("{}", x.0), x.1, color)
+            });
             if self.languages.len() > 6 {
                 let mut languages = iter.by_ref().take(6).collect::<Vec<_>>();
                 let other_sum = iter.fold(0.0, |acc, x| acc + x.1);
-                languages.push(("Other".to_owned(), other_sum));
+                languages.push(("Other".to_owned(), other_sum, Color::White));
                 languages
             } else {
                 iter.collect()
             }
         };
 
+        for (_, language) in languages.iter().enumerate() {
+            let bar_width = std::cmp::max(
+                (language.1 / 100. * language_bar_length as f64).round() as usize,
+                1,
+            );
+            language_field.push_str(&format!(
+                "{:<width$}",
+                "".on_color(language.2),
+                width = bar_width
+            ));
```

Comment:
Instead of pushing strings in a loop, you might want to map `languages.iter()` to an iter of `String`s and then join/collect them. Otherwise, we have a rough idea of how many characters are needed (`language_bar_length` + ANSI escapes + extra characters depending on `std::cmp::max`), so we might want to use `String::with_capacity` to prevent reallocations.

Reply:
cf. 72aa83c, is this what you had in mind ?

Reply:
Sorry, I meant using `with_capacity` if *not* collecting. AFAIK Rust's compiler is pretty smart about collecting lazy iterators of strings, and will efficiently allocate the necessary memory. https://github.com/o2sh/onefetch/pull/585/commits/64929ef226173013a90bddf519e25c2ece2b9889 is an example of what I was thinking.